File: pset6/vigenere/vigenere.py
import sys
import cs50

# check for a single key
if len(sys.argv) != 2:
    print("ERROR: Must enter a single key to continue.")
    exit(1)

# check if key entered is alphabetical
elif not (sys.argv[1].isalpha()):
    print("ERROR: Must enter a key with alphabet only to continue.")
    exit(1)

else:
    keys = []
    for key in sys.argv[1]:
        if key.isupper():
            keys.append(ord(key) - 65)
        elif key.islower():
            keys.append(ord(key) - 97)

    p = cs50.get_string("plaintext: ")

    j = 0
    c = ""

    for i in range(len(p)):
        # j index for key list is reset once it reaches the length of the list
        if j == len(keys):
            j = 0
        if p[i].islower():
            c += chr((ord(p[i]) - 97 + keys[j]) % 26 + 97)
            j += 1
        elif p[i].isupper():
            c += chr((ord(p[i]) - 65 + keys[j]) % 26 + 65)
            j += 1
        else:
            c += p[i]

    print("ciphertext: {}".format(c))

File: pset2/vigenere/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{
    // checks for a single key
    if (argc != 2)
    {
        printf("ERROR: Must enter a single key to continue.\n");
        return 1;
    }
    else
    {
        //set key array
        int keys[strlen(argv[1])];
        int k;
        for (int j = 0; j < strlen(argv[1]); j++)
        {
            k = (int) argv[1][j];
            // if character in the key is not alphabetical, throw an error msg.
            if (k > 122 || k < 65 || (k > 90 && k < 97))
            {
                printf("ERROR: Must enter a single key with alphabet only to continue.\n");
                return 1;
                break;
            }
            else if (k > 96)
            {
                keys[j] = k - 97;
            }
            else
            {
                keys[j] = k - 65;
            }
        }


        string p = get_string("plaintext: ");
        printf("ciphertext: ");
        int ci, pi;
        int j = 0;
        char c;
        for (int i = 0; i < strlen(p); i++)
        {
            // j index for key array is reset if it exceeds the length of the array
            if (j == sizeof(keys) / sizeof(int))
            {
                j = 0;
            }
            pi = (int) p[i];
            if (pi > 96 && pi < 123)
            {
                pi = pi - 97;
                ci = (pi + keys[j]) % 26 + 97;
                c = (char) ci;
                printf("%c", c);
                j++;
            }
            else if (pi > 64 && pi < 91)
            {
                pi = pi - 65;
                ci = (pi + keys[j]) % 26 + 65;
                c = (char) ci;
                printf("%c", c);
                j++;
            }
            else
            {
                printf("%c", p[i]);
            }
        }
        printf("\n");
    }
}